auth/rpc.go

```diff
@@ -32,17 +32,31 @@ var (
 )
 
 func BuildRPCHeader() ([]byte, error) {
-	// get current host token
-	token, err := AuthTokenNonBlocking()
+	var (
+		token        string
+		err          error
+		err2         error
+		signAsMaster bool
+	)
 
+	// get current host token
+	signAsMaster = false
+	token, err = AuthTokenNonBlocking()
 	if err != nil {
-		return nil, err
+		// We may be an un-added master
+		token, err2 = MasterToken()
+		if err2 != nil {
+			log.WithError(err2).Debug("Unable to retrieve master token")
```

Comment:
I'll add an additional debug log above the if to log the error returned from trying to get the delegate token